rust/scx_utils/src/topology.rs

```diff
@@ -314,6 +355,15 @@ impl Topology {
         &self.cores
     }
 
+    /// Get a vec of all GPUs on the hosts.
+    pub fn gpus(&self) -> Vec<&GPU> {
+        let mut gpus = vec![];
+        for node in &self.nodes {
+            gpus.extend(node.gpus.values().clone());
+        }
+        gpus
```

Comment:
It *can* be a bit misleading to give out vector which isn't indexed by IDs for entities w/ IDs. Maybe provide an `iter` or return `BTreeMap` instead? The IDs are supposed to be unique system-wide, right?

Reply:
> The IDs are supposed to be unique system-wide, right?

I wasn't 100% sure on that, especially if a system has a mix of NVIDIA/AMD GPUs. For the NVIDIA case it's the same id that is used by [`device_by_index`](https://docs.rs/nvml-wrapper/latest/nvml_wrapper/struct.Nvml.html#method.device_by_index). Maybe it's better to use the PCIe bus id instead? I think it would still work with the [NVL helpers](https://docs.rs/nvml-wrapper/latest/nvml_wrapper/struct.Nvml.html#method.device_by_pci_bus_id) as well. My thought is that any scheduler specific use cases that need extra data should still be able to lookup the device by the id.

Reply:
Hmmm.... it does `node_gpus.insert(gpu.id(), gpu.clone());`, so it does assume the the ID is unique at least in the node. If nvidia/amd may overlap, maybe the ID should be an enum - ie. `Vendor(u64)`? PCI ID is fine too but can be a bit unwieldy.

Reply:
I like the enum idea, will try that out!

> so it does assume the the ID is unique at least in the node

Yeah, wasn't sure on that either. There's not great AMD libraries it seems, so maybe this is a problem for the future, but should try to do it right the first time.